### PR TITLE
FA Testing utility tool

### PR DESCRIPTION
**Pull request by Crotalus**
Basic FA Testing (FAT) utility script, useful to quickly launch a branch, tag or even a pull request from github.com/FAForever.

Requires git-bash (https://git-scm.com/download/win)

Syntax:

```
To launch current official FAF version
./fat master

To launch a specific pull request:
./fat pr-1527

To launch latest develop:
./fat develop
```

### CODE REVIEW

File: fat [hunk 1]
```diff
@@ -0,0 +1,196 @@
+#!/bin/bash
+
+LOG_PATH="C:/temp/supcom.log"
+FA_PATH="C:/ProgramData/FAForever/bin/ForgedAlliance.exe"
+
+UPSTREAM_NAME="fat-upstream"
+UPSTREAM_URI="git://github.com/FAForever/fa.git"
```
Comment (aeoncleanse) [inline fat]: URI not URL?

**Comment by aeoncleanse** [comment]
To clarify, this launches you straight into a game, it doesn't let you go through lobby. So can't be used for lobby change testing. For other stuff though, this is utterly fantastic.

@Sheeo @duk3luk3 @Uveso Can I get you to review this before merge? I am too, but this code is significantly beyond my normal level of scripting,

**Comment by aeoncleanse** [comment]
@Crotalus can we get this launching with cheats on please?

**Comment by Crotalus** [comment]
Yeah, noticed that cheating was disabled, even though it says "Cheating enabled" when checking the options in-game. Probably something with the game being launched in single-player / campaign mode.

I needed to add this at the top of game.prefs to get it working:
`
debug = {
    enable_debug_facilities = true
}
`

Maybe should let the script do that automatically if not already there (backup the prefs before touching it ofc)

**Comment by aeoncleanse** [comment]
Definitely. Can you find a way to do this please.

**Comment by Crotalus** [comment]
The option is now appended if not already there.

**Comment by Crotalus** [comment]
A nice side-effect of `enable_debug_facilities` is that you can have several ForgedAlliance.exe running at the same time.

**Comment by aeoncleanse** [comment]
You've massively affected SinglePlayerLaunch. Is this code even called when running the game through Multiplayer, as in FAF?

File: fat (hunk 1 above)
Comment (ChrisKitching) [inline fat]: Yes.

A URI is an address to a resource of some sort. A URL is specifically a web address.

**Pull request closed without merging**